lib/runtime/adapter-read.js

```diff
@@ -69,6 +69,11 @@ class AdapterRead {
     // Common read options that should be supported by all adapters.
     static get commonOptions() { return ['from', 'to', 'last', 'timeField', 'every', 'lag']; }
 
+    // subclasses can override this method to return an array
+    // if they do we'll check that all options passed to the
+    // proc are in the array
+    allowedOptions() { return false; }
```

Comment:
It seems like this would be cleaner if it were implemented as a static property in the same way that `commonOptions` is handled. That way the read implementation could validate the options without needing to actually instantiate the proc.

We could actually change `commonOptions` to be the default implementation of `allowedOptions` so that if an adapter just wants to support the common ones, they don't have to do anything, but if they want to extend then they would do something like:

```
static get allowedOptions() { return AdapterRead.allowedOptions.concat(['myOption']); }
```